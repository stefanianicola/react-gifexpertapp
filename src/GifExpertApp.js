import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

function GifExpertApp({ defaultCategories = [] }) {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            {
                categories.map(category =>
                    <GifGrid
                        key={category}
                        category={category} />
                )
            }


        </>
    )
}

export default GifExpertApp