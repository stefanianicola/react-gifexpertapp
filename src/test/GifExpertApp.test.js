import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('pruebas en GifExpertApp', () => {

    it('debe mostrar bien el componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    it('debe mostrar una lista de categorias', () => {
        const categories = ['Dragol Ball', 'Ricky and Morty']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    });

})