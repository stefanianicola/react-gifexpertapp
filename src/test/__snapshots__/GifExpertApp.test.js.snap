// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp debe mostrar bien el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`pruebas en GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Dragol Ball"
    key="Dragol Ball"
  />
  <GifGrid
    category="Ricky and Morty"
    key="Ricky and Morty"
  />
</Fragment>
`;
